estimagic/logging/update_database.py

```diff
@@ -0,0 +1,119 @@
+"""Functions to update a DataBase in a thread safe way.
+
+All write operation to a database in estimagic should be done via functions from this
+module.
+
+Public functions in this module should not require any knowledge of sqlalchemy or
+sql in general. This is also the reason why _execute_write_statements is not a public
+function.
+
+"""
+import datetime
+import pickle
+import warnings
+from pathlib import Path
+
+import pandas as pd
+import sqlalchemy
+
+
+def append_rows(database, tables, rows):
```

Comment:
This is of course partially a matter of taste but to me it would feel more intuitive and less error-prone to pass a dictionary to this with the `tables` as keys and the `rows` as values rather than two lists. 

Also, I think it would be even more user friendly if instead of a `MetaData` object a path could be supplied. The way it is now, I think people may be confused whether they have to save the database to disk after using `append_rows` or whether they have to "close" it (just like files should be closed after being opened).

Reply:
Most database functions have `database` as their first argument and `table` or `tables` as the second one. I agree that the dictionary might be more intuitive in isolation, but the current implementation is more consistent with the other functions. 

Regarding the path: No, for the following reasons:
1. This would introduce a lot of overhead and it is really not how databases are meant to be used. 
2. I find it quite intuitive that there are `prepare_database` and `load_database` functions that both give you exactly what you need to use any function with a database argument. 
3. I know that people will worry how often the database is synchronized with the file on disk. I get this question all the time. (The answer is whenever we do an update). But this is not the place to document it or signal it via an interface. No user will see or use those functions directly. I will let you do a review of the user documentation on the logging when I wrote it and then you can decide if it is clear enough.